### PR TITLE
Remove files current subsetdef issue.2

### PR DESCRIPTION
**Pull request by buthanoid**
See https://github.com/JMMC-OpenDev/oiexplorer/issues/2
See https://github.com/JMMC-OpenDev/oiexplorer-core/pull/11

Note that the second test fails. Question: Is the target manager supposed to keep old targets ?

### CODE REVIEW

**Comment by gmella** [comment]
Seems to do the job on my first tries ! We may display the target names in addition or instead the number of target per files. :+1:

File: src/main/java/fr/jmmc/oiexplorer/gui/action/RemoveAction.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    /**
+     * Class logger
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(className);
+
+    /**
+     * Action to to remove from collection all OIFitsFiles that are concerned by the current SubsetDefinition. This
+     * Action is used in ApplicationData.
+     */
+    public final static String ACTION_CURRENT_SUBSET_DEF = "currentSubsetDefinition";
```
Comment (bourgesl) [inline src/main/java/fr/jmmc/oiexplorer/gui/action/RemoveAction.java]: use actionName like other actions

File: src/main/java/fr/jmmc/oiexplorer/gui/action/RemoveAction.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    /**
+     * Handle the action event
+     *
+     * @param evt action event
+     */
+    @Override
+    public void actionPerformed(final ActionEvent evt) {
+        if (action == null) {
```
Comment (bourgesl) [inline src/main/java/fr/jmmc/oiexplorer/gui/action/RemoveAction.java]: directly delegate to OIFitsFileListPanel action

File: src/main/java/fr/jmmc/oiexplorer/gui/action/RemoveAction.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            default:
+                LOGGER.error("unknown action.");
+                break;
+        }
+    }
+
+    private void actionCurrentSubsetDefinition() {
+        OIFitsFileListPanel oiFitsFileListPanel = OIFitsExplorer.getInstance().getMainPanel().getOIFitsFileListPanel();
+
+        List<OIFitsFile> selecteds = oiFitsFileListPanel.getSelectedOIFitsFiles();
```
Comment (bourgesl) [inline src/main/java/fr/jmmc/oiexplorer/gui/action/RemoveAction.java]: Move all the code in OIFitsFileListPanel removeSelectedFiles() with the confirm dialog
TODO: fix the message to show target names

File: src/main/java/fr/jmmc/oiexplorer/gui/OIFitsFileListPanel.java [hunk 4]
```diff
@@ -367,4 +368,34 @@ public String getToolTipText(final MouseEvent evt) {
 
         return list;
     }
+
+    /** Returns the list of selected Values. which is equivalent to the files concerned by the current subset definition.
+     *
+     * @return the list of selected values.
+     */
+    public List<OIFitsFile> getSelectedOIFitsFiles() {
```
Comment (bourgesl) [inline src/main/java/fr/jmmc/oiexplorer/gui/OIFitsFileListPanel.java]: private

File: src/main/java/fr/jmmc/oiexplorer/gui/OIFitsFileListPanel.java [hunk 5]
```diff
@@ -367,4 +368,34 @@ public String getToolTipText(final MouseEvent evt) {
...
+     * This list of files is exactly the current selection of this.oifitsFileList.
+     * @return the list of removed files, without the (potential) null failure values.
+     */
+    public List<OIFitsFile> removeSelectedOIFitsFiles() {
+        List<OIFitsFile> filesToRemove = this.oifitsFileList.getSelectedValuesList();
+
+        List<OIFitsFile> filesRemoved = ocm.removeListOIFitsFile(filesToRemove);
+
+        // remove nulls
+        ListIterator<OIFitsFile> iterator = filesRemoved.listIterator();
```
Comment (bourgesl) [inline src/main/java/fr/jmmc/oiexplorer/gui/OIFitsFileListPanel.java]: just compare both lists to report files that were not removed ?

**Review by bourgesl — COMMENTED** [review]
Minor changes needed.
I will look at the Fest test code to fix it (asynchronous processing)

**Comment by bourgesl** [comment]
Fest test fixed and passed.
See m2_addTwoFilesTwoTargetRemoveOneTarget : target manager returns two targets (Gam Vic, UNDEFINED)

**Pull request merged**